src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -905,15 +908,20 @@ private Task DestroyActivations(List<ActivationData> list)
                         logger.Warn(ErrorCode.Catalog_UnregisterMessageTarget3, String.Format("Last stage of DestroyActivations failed on {0}.", activationData), exc);
                     }
                 }
+                logger.Info(ErrorCode.Catalog_DestroyActivations_Done, "Done FinishDestroyActivations #{0} - Destroyed {1} Activations.", number, list.Count);
+                
+            }
+            catch (Exception exc)
+            {
+                logger.Error(ErrorCode.Catalog_FinishDeactivateActivation_Exception, String.Format("FinishDestroyActivations #{0} failed with {1} Activations.", number, list.Count), exc);
+            }
+            finally
+            {
                 // step 5 - Resolve any waiting TaskCompletionSource
                 if (tcs != null)
                 {
-                    tcs.SetMultipleResults(list.Count);
+                    tcs.TrySetMultipleResults(list.Count);
```

Comment:
Basically, just moved this tcs.TrySetMultipleResults to be in Finally, just to be 100% sure we always resolve it.